tests/virt_autotest/libvirt_extend_storage_lvm.pm

```diff
@@ -0,0 +1,121 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Virtualization Extend LVM Storage pool / volume test
+#
+# Test flow:
+# - Check a additional free hard disk
+# - Wipe hard disk clean
+# - Partition test disk
+# - Create a pv and display result
+# - Create a vg named 'lvm_vg' and display result
+# - Create a storage pool (lv) named 'guest_image_lvm'
+# - Create a logical volume (lv) size 1G
+# - Attach a logical volume (lv) to guest systems
+# - Clone a logical volume (lv)
+# - Cleanup
+# Maintainer: Leon Guo <xguo@suse.com>
+
+use base "virt_feature_test_base";
+use virt_autotest::virtual_storage_utils;
+use virt_autotest::utils;
+use virt_autotest::common;
+use strict;
+use warnings;
+use testapi;
+use utils;
+use virt_utils;
+use version_utils 'is_sle';
+
+our $lvm_vg_name = 'lvm_vg';
+our $lvm_pool_name = 'guest_image_lvm';
+our $timeout = 180;
+sub run_test {
+    my ($self) = @_;
+
+    record_info "Prepare Guest Systems";
+    foreach (keys %virt_autotest::common::guests) {
+        start_guests() unless is_guest_online($_);
+    }
+
+    ## About LVM volumes management
+    my ($dev, $lvm_disk);
+    # Check with all existed Hard disks
+    my @disks = split(/\n/, script_output("lsblk -n -l -o NAME -d -e 7,11"));
+    my $scalar = @disks;
+    # Requires at least 2 physical hard disks for LVM Storage test
+    if (($scalar eq 1) || get_var('KEEP_DISKS')) {
+        record_info("WARNING", "Requires at least 2 physical hard disks for LVM Storage test\n", result => 'softfail');
+        return;
+    }
+    # Use a free(unused) hard disk for LVM volumes
+    $dev = "/dev/";
+    foreach my $disk (@disks) {
+        if (script_run("set -o pipefail;lsblk -rnoPKNAME,MOUNTPOINT | grep -i $disk | awk \'{print \$2}\'") ne '0') {
+            $lvm_disk = $dev . $disk;
+            last;
+        }
+    }
+    record_info "Assign a New Disk:", "$lvm_disk";
+    # Wipe Hard Disk Clean via dd for assigned a new full disk
+    wipe_hard_disk($lvm_disk);
+    # Create a Volume Group (VG) with LVM named 'lvm_vg'
+    create_volume_group($lvm_disk);
```

Comment:
If line 43~65 can be wrapped into a function like prepare_storage_source, it will be much clearer. HDYT?

Reply:
Yes, good idea. create a new fuc as `prepare_lvm_storage_pool_source`
Add the latest validation run:
[gi-guest_developing-on-host_sles12sp5-kvm](http://openqa.suse.de/tests/9312020#)

Reply:
Code not pushed yet?

Reply:
Create a new fuc as `prepare_lvm_storage_pool_source`. figure out the unused hard disk for lvm volume. 
keep `wipe_hard_disk($lvm_disk)`; and `create_volume_group($lvm_disk)`;
look to **## Cleanup** step -> `delete_physical_volume($lvm_disk)`; and `wipe_hard_disk($lvm_disk)`;
Have pushed the latest code for this pr already. please check code again. thanks.

Reply:
@guoxuguang Hi Leon, I am afraid you misunderstood something here. prepare_lvm_storage_pool_source is intended to be more than finding an unused disk that it should include also wipe_hard_disk and create_volume_group. That is any step making the source of the storage pool ready should be in -- the source used in `virsh pool-define-as $lvm_pool_name logical --source-name $lvm_vg_name ...`. That's how this func name is originated from, which is potentially to be inherited and overwritten by other similar storage tests.

Reply:
yes, totally make sense. try to merge `wipe_hard_disk` and `create_volume_group` into `prepare_lvm_storage_pool_source`. during **cleanup** step,  `delete_physical_volume` and `wipe_hard_disk ` required `$lvm_disk` was defined from `prepare_lvm_storage_pool_source`. So, try to figure out a unused hard disk as $lvm_disk for lvm volume from fuc prepare_lvm_storage_pool_source.  and keep `wipe_hard_disk` and `create_volume_group` out of `prepare_lvm_storage_pool_source`.

Reply:
@alice short update,  update func `prepare_lvm_storage_pool_source`. and merge `wipe_hard_disk` and `create_volume_group` into `prepare_lvm_storage_pool_source` now.